### ISSUE TITLE
[FP] Unnecessary type casting

### ISSUE THREAD

**Issue by orklah** [kalessil/phpinspectionsea#1216]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         |  Php Inspections (EA Ultimate),2.0.13|

Current behaviour
--
I came across something like this today:
```php
    $a = 3;
    if($a === 1) {
        $b = 1;
    } elseif($a === 2) {
        $b = 2;
    }

    $c = (int)$b;
```

The last line triggers 2 inspections:
- Unnecessary type casting
- Variable $b might not be defined

Which means that if I want $c to be int only, the type casting is necessary, or else, $c will be int|null

Thanks

**Comment by funivan** [comment]
I do not think that we can fix this issue. 
@orklah what is your solution? You need to fix this code because of undefined variable =)

**Comment by orklah** [comment]
> I do not think that we can fix this issue.
> @orklah what is your solution? You need to fix this code because of undefined variable =)

Probably. I could fix it by adding $b = null; at the top of everything but this is not what bothers me. 

If I apply the fix on the wrong inspection (unneccessary casting) I will break functionnal code by changing $b type.

Maybe there is nothing we can do, maybe we should just add a warning or change the label to something like "$b seems to be only int, the cast could be removed". It could at least warn that there may be a danger

**Comment by kalessil** [comment]
The fix will probably break the code, but we don't have tooling to recognize if the variable is always initialized or not. Hence declining.
